debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSnapshotChangeEventSource.java

```diff
@@ -6,229 +6,144 @@
-        this.previousOffset = previousOffset;
         this.jdbcConnection = jdbcConnection;
-        this.schema = schema;
     }
 
     @Override
-    public SnapshotResult execute(ChangeEventSourceContext context) throws InterruptedException {
-        // for now, just simple schema snapshotting is supported which just needs to be done once
-        if (previousOffset != null) {
-            LOGGER.debug("Found previous offset, skipping snapshotting");
-            return SnapshotResult.completed(previousOffset);
+    protected SnapshotContext prepare(ChangeEventSourceContext context) throws Exception {
+        if (connectorConfig.getPdbName() != null) {
+            jdbcConnection.setSessionToPdb(connectorConfig.getPdbName());
         }
 
-        Connection connection = null;
-        SnapshotContext ctx = null;
-
-        try {
-            connection = jdbcConnection.connection();
-            connection.setAutoCommit(false);
-
-            if (connectorConfig.getPdbName() != null) {
-                jdbcConnection.setSessionToPdb(connectorConfig.getPdbName());
-            }
-
-            ctx = new SnapshotContext(
-                    context,
-                    connection,
-                    connectorConfig.getPdbName() != null ? connectorConfig.getPdbName() : connectorConfig.getDatabaseName()
-            );
-
-            determineCapturedTables(ctx);
-
-            if (!lockTablesToBeCaptured(ctx)) {
-                return SnapshotResult.aborted();
-            }
-
-            determineOffsetContextWithScn(ctx);
-            readTableStructure(ctx);
-
-            if (!createSchemaChangeEventsForTables(ctx)) {
-                return SnapshotResult.aborted();
-            }
-
-            return SnapshotResult.completed(ctx.offset);
-        }
-        catch(RuntimeException e) {
-            throw e;
-        }
-        catch(Exception e) {
-            throw new RuntimeException(e);
-        }
-        finally {
-            if (ctx != null) {
-                ctx.dispose();
-            }
-
-            rollbackTransaction(connection);
+        return new OracleSnapshotContext(
+                connectorConfig.getPdbName() != null ? connectorConfig.getPdbName() : connectorConfig.getDatabaseName()
+        );
+    }
 
-            if (connectorConfig.getPdbName() != null) {
-                jdbcConnection.resetSessionToCdb();
-            }
-        }
+    @Override
+    protected Set<TableId> getAllTableIds(SnapshotContext ctx) throws Exception {
+        return jdbcConnection.readTableNames(ctx.catalogName, null, null, new String[] {"TABLE"} );
     }
 
-    private void determineCapturedTables(SnapshotContext ctx) throws SQLException {
-        Set<TableId> allTableIds = jdbcConnection.readTableNames(ctx.catalogName, null, null, new String[] {"TABLE"} );
+    @Override
+    protected boolean lockTables(ChangeEventSourceContext sourceContext, SnapshotContext snapshotContext) throws SQLException {
```

Comment:
Would it be possible to pull this method up and introduce an abstract method lockTable()?